app/services/courses/generate_course_code_service.rb

```diff
@@ -0,0 +1,17 @@
+module Courses
+  class GenerateCourseCodeService
+    def execute
+      "#{valid_letters.sample}#{3.times.map { valid_number }.join}"
+    end
+
+  private
+
+    def valid_letters
+      ('A'..'Z').to_a - %w[O I]
+    end
+
+    def valid_number
+      (0..9).to_a.sample.to_s
```

Comment:
0..999 with padding?

Reply:
For those wondering how, I googled for it:

```ruby
> 99.to_s.rjust(3, "0")
=> "099"
> "%03d" % 99
>= "099"
```

Reply:
you beat me

Reply:
Done. Thanks 👍